### PR TITLE
Added auto-closing functionality

### LINKED ISSUE #83: Support System Auto-Close
**Acceptance Criteria**

* After 2 days of no messages, Chat Bot should warn a channel that it will be automatically closed.
  * If the last message sent in the chatroom after 2 days have passed is chat bot's warning to close, Chat Bot should automatically close the support room.

**Suggested Technical Implementation**

Chat Bot should run a job hourly in which it checks through all the support chatrooms to see if the last message in that room is over 24*3 hours old.  If so, and the last message is from Chat Bot, the room should be automatically closed.  Otherwise, if the message is not from chat bot (and still meets the age requirement), Chat Bot should post a warning that the room will be automatically closed in 3 days.

### PR DESCRIPTION
**Pull request by McJeffr**
Additionally, cleaned up some duplicate code that would be created as a result of this new feature.

The new configuration options are as follows:
```
    support: {
        autoClose: {
            warning: 2 * 24 * 60 * 60 * 1000,
            closing: 2 * 24 * 60 * 60 * 1000
        }
    },
```
Here, `warning` refers to the amount of time to pass by before a warning notice is being sent to the channel. This means that if the previous message was `warning` ms ago, a warning notice will be posted. This timer "resets" when someone else posts a message in the channel. The `closing` refers to the amount of time to pass by **after the warning message** in ms before the channel gets automatically closed. If someone posts a message, this timer "resets" as the last message must be the warning notice for this to take affect.

### CODE REVIEW

File: src/Command/Support.js [hunk 1]
```diff
@@ -71,10 +63,10 @@ module.exports = Command.extend({
             }
 
             if (tokens[0].toLowerCase() === 'close') {
-                return await this.processDeletion(message, commandParameters);
```
Comment (McJeffr) [inline src/Command/Support.js:74]: Unsure what "commandParameters" even did here, the `processDeletion` function didn't even require this.

File: config/default.js [hunk 2]
```diff
@@ -105,6 +105,10 @@ module.exports = {
                 description: "Support for MineZ: Stuck in a block; Broken dungeons; Death due to cheaters, bugs, and broken dungeons.",
                 roles: ['618856455206076417']
             }
+        },
+        autoClose: {
+            warning: 2 * 24 * 60 * 60 * 1000,
```
Comment (navarr) [inline config/default.js]:
```suggestion
            warning: 1 * 24 * 60 * 60 * 1000,
```

File: src/Utility/SupportCloser.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        'i18n': 'i18n',
+        'moment': 'moment'
+    },
+    initialize: function (dependencyGraph) {
+        this._super(dependencyGraph);
+
+        this.discordClient.once("ready", () => {
+            const supportCategory = this.retrieveSupportCategory();
+            if (!supportCategory || !supportCategory.children) {
+                console.log("WARNING: could not retrieve support category for automatic support closer utility");
```
Comment (navarr) [inline src/Utility/SupportCloser.js]:
```suggestion
                console.warn("Warning: Could not retrieve support category for automatic support closer utility");
```

File: src/Command/Support.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
-const crypto = require('crypto');
 const config = require('config'); // Explicit import is necessary here for command aliases
 
 const Command = require('../Command');
 const RoleDeterminer = require('../Helper/RoleDeterminer');
-const timeoutPromise = require('../Helper/TimeoutPromise');
-const getChannelLog = require('../Helper/ChannelLogger');
+const { generateRoomName, parseRoomType, convertRoom, archiveRoom } = require('../Helper/SupportRoomHelper');
```
Comment (navarr) [inline src/Command/Support.js]: I think only archiveRoom is shared right?  Why not have it be its own helper (SupportRoomArchiver) and leave the others as functions in this file?

IMO, SupportRoomHelper is violating the SRP

**Review by navarr — CHANGES_REQUESTED** [review]

File: src/Command/Support.js (hunk 4 above)
Comment (McJeffr) [reply src/Command/Support.js]: parseRoomType and getRoomParticipants are used by archiveRoom. The archiving of a room uses these two functions. I could move it to a SupportRoomArchiver, but there won't really be any functions left in SupportRoomHelper other than the name conversion (which uses generateRoomName). I don't think the SRP is broken here, these are all functions directly involved with support rooms (either archiving or converting, but the functions used for these two are shared among each of these two functions).

Comment (navarr) [reply src/Command/Support.js]: If parseRoomType and getRoomParticipants are used by both archiveRoom and the Support command general functionality, they should be split out into their own classes/files.

What you've done here is create a container for support-related commands, and while that helps clean up the Support command file, by itself its messy.  There's no harm in breaking these out into their own files, which is what I'd recommend here.

* SupportRoomArchiver
* SupportRoomConverter
* SupportRoomTypeParser
* SupportRoomNameGenerator

All of those would serve a single purpose that its name clearly details, and you wouldn't have weird reverse mapping like `const { a, b, c }`

Comment (McJeffr) [reply src/Command/Support.js]: Alright, that could work too. I'll look into it when I have time later today.

**Review by navarr — APPROVED** [review]

**Pull request merged**